openssl/src/x509/mod.rs

```diff
@@ -548,6 +604,21 @@ impl X509Ref {
         }
     }
 
+    /// Returns certificate version. If this certificate has no explicit version set, it defaults to
+    /// version&nbsp;1.
+    ///
+    /// Note that `0` return value stands for version 1, `1`&nbsp;&ndash; for version 2 and so on.
+    ///
+    /// This corresponds to [`X509_get_version`].
+    ///
+    /// [`X509_get_version`]: https://www.openssl.org/docs/man1.1.1/man3/X509_get_version.html
+    #[cfg(ossl110)]
+    pub fn version(&self) -> i32 {
+        // Covered with `x509_ref_version()`, `x509_ref_version_no_version_set()`,
+        // `x509_ref_version_incorrect_version_set()` tests
+        unsafe { ffi::X509_get_version(self.as_ptr()) as i32 }
```

Comment:
In order `version()` method to work on both x32 and x64 OSs a cast to i32 is performed. Is this appropriate solution or should there be some kind of check performed in order to determine which `c_long` type definition was taken: from `libc-0.2.76/src/unix/linux_like/linux/gnu/b64/x86_64/not_x32.rs` or `libc-0.2.76/src/unix/linux_like/linux/gnu/b64/x86_64/x32.rs`?

Reply:
This is fine.